### ISSUE TITLE
Parentheses in the expressions are necessary

### ISSUE THREAD

**Issue by EugineKosenko** [skeeto/emacsql#9]
It seems like a bug in the query compiler. The next code:

<pre><code>
(emacsql-compile storage
 [:select *
  :from atable
  :where (and (= field1 1)
          (or (= field2 2)
          (= field3 3)))])
</code></pre>

returns the result

<code>"SELECT \* FROM atable WHERE field1 = 1 AND field2 = 2 OR field3 = 3;"</code>

while the next statement is expected:

<code>"SELECT \* FROM atable WHERE field1 = 1 AND (field2 = 2 OR field3 = 3);"</code>

Actually, there must be parentheses here.

**Comment by skeeto** [comment]
Yup, this is a bug in the query compiler. It doesn't understand operator
precedence and just dumps everything out flat. There should be a bunch
of similar cases with the same problem.

Perhaps it should always use parentheses just to be sure?

**Comment by EugineKosenko** [comment]
> Perhaps it should always use parentheses just to be sure?

I think so. The compiled output is not for a human eyes, and there is almost no matter how ugly the output is. I would remove the parentheses manually in the rare debug cases.

**Comment by skeeto** [comment]
Working on adding parentheses around everything. I need to figure out a
problem with a syntax shortcut I took early on: what follows an "UPDATE
... SET" isn't a direct expression like I'm pretending, so it doesn't
tolerate being wrapped in parentheses. It's "column = <expression>",
making it another special case similar to "INSERT ... VALUES". 

Otherwise it seems to be working fine.

**Comment by tarsius** [comment]
@skeeto Could you please have another look at this? Nesting `or`s and `and`s would make some things much easier. Thanks!

**Comment by ak-coram** [comment]
@tarsius: I'm hoping this is now solved with a164ecd9d31dbd45a646244906ac077d5a331419:

```elisp
[:select *
 :from atable
 :where (and (= field1 1)
             (or (= field2 2)
                 (= field3 3)))]
; => "SELECT * FROM atable WHERE field1 = 1 AND (field2 = 2 OR field3 = 3);"
```

**Comment by skeeto** [comment]
Should now be resolved by #47 and #48.

**Comment by tarsius** [comment]
Thanks!
